### PR TITLE
Cleanup package selection

### PR DESCRIPTION
**Pull request by alcohol**

### CODE REVIEW

File: src/PackageSelection/PackageSelection.php [hunk 1]
```diff
@@ -490,15 +503,7 @@ private function matchAddr($addr1, $addr2, $len = 0, $chunklen = 32)
         return true;
     }
 
-    /**
-     * Add repositories to a pool
-     *
-     * @param Pool  $pool
-     *  The Pool instance
-     * @param RepositoryInterface[] $repos
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: I would keep this phpdoc, as it is more precise.

File: src/PackageSelection/PackageSelection.php [hunk 2]
```diff
@@ -525,25 +527,22 @@ private function setSelectedAsAbandoned()
         }
     }
 
-    /**
-     * Gets a list of filtered Links.
-     *
-     * @param Composer $composer The Composer instance
-     *
-     * @return Link[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: this should be kept, as it is more precise than `array`

File: src/PackageSelection/PackageSelection.php [hunk 3]
```diff
@@ -552,40 +551,41 @@ private function getFilteredLinks(Composer $composer)
      *
      * This method is called when 'require-all' is set to true.
      *
-     * @param array  $repos            List of all Repositories configured
+     * @param array  $repositories            List of all Repositories configured
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: alignment should be fixed

File: src/PackageSelection/PackageSelection.php [hunk 4]
```diff
@@ -692,121 +693,94 @@ private function selectLinks(Pool $pool, $links, bool $isRoot, bool $verbose)
         return $depsLinks;
     }
 
-    /**
-     * Create the additional repositories
-     *
-     * @return RepositoryInterface[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: should be kept

File: src/PackageSelection/PackageSelection.php [hunk 5]
```diff
@@ -692,121 +693,94 @@ private function selectLinks(Pool $pool, $links, bool $isRoot, bool $verbose)
...
-        return $depRepos;
+        return $repositories;
     }
 
-    /**
-     * Gets All or filtered Packages of a Repository.
-     *
-     * @param RepositoryInterface $repo a Repository
-     *
-     * @return PackageInterface[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: should also be kept

File: src/PackageSelection/PackageSelection.php [hunk 6]
```diff
@@ -692,121 +693,94 @@ private function selectLinks(Pool $pool, $links, bool $isRoot, bool $verbose)
...
     }
 
-    /**
-     * Gets the required Links if needed.
-     *
-     * @param PackageInterface $package A package
-     * @param bool $isRoot
-     *  If the packages are linked in root or as dependency
-     *
-     * @return Link[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: should be kept

File: src/PackageSelection/PackageSelection.php [hunk 7]
```diff
@@ -692,121 +693,94 @@ private function selectLinks(Pool $pool, $links, bool $isRoot, bool $verbose)
...
 
         return $required;
     }
 
-    /**
-     * Filter given repositories.
-     *
-     * @param RepositoryInterface[] $repositories
-     *
-     * @return RepositoryInterface[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: should be kept

File: src/PackageSelection/PackageSelection.php [hunk 8]
```diff
@@ -692,121 +693,94 @@ private function selectLinks(Pool $pool, $links, bool $isRoot, bool $verbose)
...
+            }
+        );
     }
 
-    /**
-     * Filter given repositories.
-     *
-     * @param RepositoryInterface[] $repositories
-     *
-     * @return RepositoryInterface[]
```
Comment (stof) [inline src/PackageSelection/PackageSelection.php]: should be kept

**Comment by alcohol** [comment]
I would really like to extract the "strip hosts" related methods to another class/helper. Any suggestions or input on this @stof ?

**Comment by alcohol** [comment]
Probably better left for another PR.

**Pull request merged**